Medical additions v2

KillerTheFlareon opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
Adds new items to medical, and reworks corpsman into a new job.
## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
Medical does not like the entire chemlocker going boom. This gives them an alternative that they can make with reinforced glass as a T2 med research (vials/bottles/jugs/beakers) which also boast a nice boost to capacity, Para gets a portable defib like the corps and i've made a new item called "hypoinjector" which is essentially a faster syringe themed as an autoinjector with a limiter (press injector against skin, hold down button to draw or press button to inject), which I'm sure med will enjoy.

I've fixed brig loadouts so all their gear now spawns properly in the locker, and also separated brig's duties into 2 roles instead of 1 due to the increased demand on highpop. There is now a new corpsman role that handles all the external duties of brig (rescuing crit/dead security officers, providing on-field medical treatment & responding to calls for medical attention over coms)

This is mainly due to the fact the brig was expected to act like a doctor/chemist/paramedic/combat med all at once, which works well enough in lower pop but results in "the brig is never in the brig" on higher pop as they're constantly running around rescuing officers or providing immediate medical attention in the field whilst prisoners die in the brig. Or officers show up to be treated only to have no brigmedic in brig.

So Corpsman takes on the para/combat med responsibilities which takes weight off of the brigmedic's shoulders allowing them to focus on prisoners/seccies in sec or just cooking chems, and also allows them to follow their sop of "you really shouldn't leave brig as brigmedic" which was previously impossible to follow due to the contradicting responsibilities. 

This also means all of the previous mobility tools (minus the brig monitor, as I don't expect mappers to start mapping in a vitals monitor to brigmed anytime soon) are removed from brig and given to corps instead (defib/big medkit & a new 10u vial using hypospray, which is also a syndicate target objective.) 

I've also made the brigmedic bedsheet a thief target, like it was intended to be but never done.
## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->
Corp & Brig side-by-side
<img width="652" height="476" alt="ytjuytfdk" src="https://github.com/user-attachments/assets/8d442085-dc7c-48d0-ad46-da386b59917d" />
It has a hood!
<img width="581" height="446" alt="tyjdutgk" src="https://github.com/user-attachments/assets/4a11b896-b0c5-485f-8505-b55c6dd121b1" />
Medical additions
<img width="497" height="360" alt="tyjddtuk" src="https://github.com/user-attachments/assets/71988e0c-a631-466d-9dd2-d89f4fa5a4c9" />

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: Killer Tamashi (The rest) & Faush (C# Code) & Salem (Brigmedic mask sprite)
- add: Added Reinforced Containers (T2 Med) [Reinforced Vial/Bottle/Small Beaker/Large Beaker/Jug]
- add: Added Automated Injection (T1 Med) [Hypoinjector, literally just a faster syringe]
- add: Corpsman Job & Items [Corps everything basically & Hypospray Kit]
- add: Corpsman Hypospray [Uses 10u vials, syndie objective]
- add: Corpsman to all highpop maps & other where applicable
- add: Secmed Key, you can guess whose headset it goes into
- add: Basic chem kit to brigmedic locker, lowpop brigmeds rejoice
- add: Roundstart defib for paramedic [Paramedic Defib]
- tweak: Renamed Brigmedic Hardsuit & Headset into "Security Medical Suit/Headset" respectively
- tweak: Resprited Brigmedic Headset into a grey/blue colorscheme to more fit in with sec headsets
- tweak: Brigmedic mask now has 5% infection resist chance
- tweak: Brigmedic jacket now has 40% infection resist chance, on par with winter coats.. Why do winter coats prevent-
- tweak: Resprited Brigmedic Icon to match with new Corps icon
- tweak: Brigmedic mask resprited to look better.
- fix: Corpsman jacket looking AWFUL
- fix: SOMEONE butchered the brigmedic locker's fills, this also fixes that.

KillerTheFlareon committed:
Add brigmed hypo+case+fix

..magboot spawn chance on brigmedic locker. Someone accidentally gave it a 0.1 chance to spawn lol.

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Round 2

KillerTheFlareon committed:
Reinforced stuff

KillerTheFlareon committed:
R.Glass recipes

You can now put reinforced glass in ANY lathe, but only the medical one has any recipes for them

Fasuh committed:
feat(Chemistry): Add logic for miniature hypospray to use inserted vials in place of caintainer.

KillerTheFlareon committed:
awawa

aaa

KillerTheFlareon committed:
Brigmedic clothing resprite & Hypoinjector sprites

Fixed geneva convention violation
Also gave the hypo a sprite so yippee

Fasuh committed:
fix(Hypospray): Remove drawing blood and from containers from brigmedic hypospray

Fasuh committed:
fix(Hypoinjector): add translations for hypoinjector

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
Adds steal objectives

- Brig bedsheet was always intended to be one, i just.
Didn't

- Hypo is syndie steal objective
self explanatory i'm sure.

KillerTheFlareon committed:
Forgot to change brigmedic defib

As well as FINALLY added para defib

KillerTheFlareon committed:
aaaaa

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Corpsman Addition

Whew. That was a lot of work.
This finally makes a role separation between corpsman and brigmedic, allowing the brigmedic to stay in brig and the corpsman to run out and perform the other half of the brigmedic's duties.

Basically, brig is doctor/chemist, stuck in brig
Corps is para/on field triage meant to retrieve & heal seccies in combat conditions

KillerTheFlareon committed:
Adds missing things.

Ye.

KillerTheFlareon committed:
Fixes

wawa

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Delete .equipped-OUTERCLOTHING.png-autosave.kra

KillerTheFlareon committed:
Update chemistry-vials.yml

KillerTheFlareon committed:
Update medical.yml

I forsee great squabbling over this if I didn't rename it

KillerTheFlareon committed:
last second addition

gives corps a medical roller bed, since, you know. Probably needs that to effectively drag dead bodies lmao.

KillerTheFlareon committed:
Salem offered to make sprites

Who am I to say no?

Fasuh committed:
feat(Chemistry): Add solution indicator on corpsman hypospray

KillerTheFlareon committed:
fixes

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

walksanatora commented:
target starlight-dev

KillerTheFlareon commented:
wrong branch moment

KillerTheFlareon committed:
Added 2 new spawns

Talked with pig about it

KillerTheFlareon commented:
uh

KillerTheFlareon commented:
AUGH

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
mfw high heels creates a merge conflict

KillerTheFlareon committed:
Chem additions

Given brigmedic's role in making chems is being solidified, I tossed basic chemistry equipment in their locker so they can do it.

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
GOD

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
YAML error not related to my PR

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

Happyrobot33 commented:
:skull:

KillerTheFlareon commented:
> 💀

WHAT DOES IT MEAAAAAN

KillerTheFlareon commented:
OH COME ON

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
so help me god if i have to solve one more merge conflict
i'm gonna cry

KillerTheFlareon commented:
God I can't believe I'm saying this- 
Do not merge until I push another update, I found a bug that needs to be fixed by faush

Happyrobot33 commented:
do we say fuck it we ball?

Happyrobot33 commented:
oh killer hasnt fixed it yet

KillerTheFlareon commented:
> do we say fuck it we ball?

i'm gonna cry, i missed a "fuck it we ball" moment from happy

(I was watching the newest digital circus episode)

KillerTheFlareon committed:
fixes

not THE fix. But uh. Yeah.

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon committed:
Update loadout_groups.yml

KillerTheFlareon commented:
ah shit

KillerTheFlareon commented:
yaml linter is not my problem

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon committed:
Update ExternalContainerInjectorSystem.cs

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

Happyrobot33 committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
welp.
That's a merge conflict with a C# File.
Fuck.

KillerTheFlareon commented:
alright it's not joever i can fix this

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon commented:
nevermind
i'm just gonna be
sitting in a corner i guess

Commit:
chore(Chemistry): Fix injector after upstream merge

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon committed:
wiz changes made somethin break

fixed

KillerTheFlareon commented:
Good to be merged. Again..

Rinary1 commented on Content.Client/Chemistry/UI/HyposprayStatusControl.cs:
```diff
@@ -50,9 +50,8 @@ protected override void FrameUpdate(FrameEventArgs args)
         });
 
-        _label.SetMarkup(Loc.GetString("hypospray-volume-label",
+        _label.SetMarkup(Loc.GetString("brigmedic-hypospray-volume-label",
```
Why? This affects on any hypospray. It's better to make `bool` in `HyposprayComponent` for disabling mode visibilty or add new string in `HyposprayComponent` with locId for volume label, so it can be with "modeString" or without.

KillerTheFlareon replied:
Not my code. Will ask the person who wrote it.

Conflee commented:
Sprites approved because we're redoing BSO and ERT Leader anyway so the colors shouldn't confuse people, given the Corpsman only uses a hint of the dark blue in a couple spots.

Rinary1 commented on Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs:
```diff
@@ -18,6 +18,7 @@
 using Content.Shared.Popups;
 using Content.Shared.Stacks;
 using Content.Shared.Verbs;
+using Robust.Shared.Timing;
```
```suggestion
using Robust.Shared.Timing; // Starlight-edit
```

Rinary1 commented on Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs:
```diff
@@ -32,6 +33,7 @@ public abstract class SharedInjectorSystem : EntitySystem
     [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
     [Dependency] private readonly SharedPopupSystem _popup = default!;
     [Dependency] private readonly SharedForensicsSystem _forensics = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
```
```suggestion
    [Dependency] private readonly IGameTiming _gameTiming = default!; // Starlight-edit
```

Rinary1 commented on Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs:
```diff
@@ -146,16 +148,19 @@ private void OnInjectDoAfter(Entity<InjectorComponent> ent, ref InjectorDoAfterE
     /// </summary>
     private void InjectDoAfter(Entity<InjectorComponent> injector, EntityUid target, EntityUid user)
     {
+        if (!_gameTiming.IsFirstTimePredicted)
+            return;
```
```suggestion
        // Starlight-start
        if (!_gameTiming.IsFirstTimePredicted)
            return;
        // Starlight-end
```

Rinary1 commented on Content.Shared/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorComponent.cs:
Tbh, i don't think so that's a good idea to create separate Component + Systems which copy Hypospray and have some unique code. It's better to merge this to Hypospray, make boolean depending on which one will be used vials or solution.

KillerTheFlareon replied:
Wouldn't that make upstream merge conflicts?

Rinary1 replied:
> Wouldn't that make upstream merge conflicts?

Maybe, but it's anyway better than make duplications

Rinary1 commented on Resources/Locale/en-US/_Starlight/chemistry/hypospray-vial-loader.ftl:
```diff
@@ -0,0 +1,4 @@
+brigmedic-hypospray-vial-slot-name = Vial Slot 
+brigmedic-hypospray-eject-vial = Eject Vial
```
```suggestion
hypospray-vial-slot-name = Vial Slot 
hypospray-eject-vial = Eject Vial
```

Rinary1 commented on Resources/Locale/en-US/_Starlight/chemistry/hypospray-vial-loader.ftl:
```diff
@@ -0,0 +1,4 @@
+brigmedic-hypospray-vial-slot-name = Vial Slot 
+brigmedic-hypospray-eject-vial = Eject Vial
+
+brigmedic-hypospray-volume-label = Volume: [color=white]{$currentVolume}/{$totalVolume}u[/color]
```
```suggestion
hypospray-nomode-volume-label = Volume: [color=white]{$currentVolume}/{$totalVolume}u[/color]
```

Rinary1 commented on Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs:
```diff
@@ -0,0 +1,90 @@
+        _prevVolume = solution.Volume;
+        _prevMaxVolume = solution.MaxVolume;
+
+        _label.SetMarkup(Loc.GetString("brigmedic-hypospray-volume-label",
```
```suggestion
        _label.SetMarkup(Loc.GetString("hypospray-nomode-volume-label",
```

Rinary1 commented on Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs:
```diff
@@ -0,0 +1,90 @@
+                _prevMaxVolume = FixedPoint2.Zero;
+            }
+
+            _label.SetMarkup(Loc.GetString("brigmedic-hypospray-volume-label",
```
```suggestion
            _label.SetMarkup(Loc.GetString("hypospray-nomode-volume-label",
```

Rinary1 commented on Resources/Locale/en-US/_Starlight/objectives/conditions/steal-target-groups.ftl:
```diff
@@ -1,5 +1,11 @@
+
+# Brigmedic items
+
+steal-target-groups-brigmedichypo = corpsman hypospray
```
```suggestion
steal-target-groups-corpsman-hypo = corpsman hypospray
```

Rinary1 commented on Resources/Prototypes/Entities/Structures/Machines/lathe.yml:
```diff
@@ -503,6 +503,8 @@
     - StarlightCyberlimbDynamic
     - StarlightCyberOrgansDynamic
     - StarlightImplantDynamic
+    - StarlightMedContainersDynamic
+    - StarlightMedSyringeDynamic
```
Add here Starlight comments:
# Starlight-start
# Starlight-end

(I can't suggest on untouched fields)

KillerTheFlareon replied:
Done

Rinary1 commented on Resources/Prototypes/Roles/Jobs/departments.yml:
```diff
@@ -116,6 +116,7 @@
   - SecurityCadet
   - SecurityOfficer
   - Brigmedic
+  - Corpsman
```
```suggestion
  - Corpsman # Starlight-edit
```

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Clothing/Eyes/glasses.yml:
```diff
@@ -23,6 +23,30 @@
+  parent: [ClothingEyesBase, ShowSecurityIcons, BaseSecurityContraband]
+  id: ClothingEyesGlassesCorpsman
+  name: corpsman glasses
+  description: Super Tamashi-Upgraded glasses. Corpsman ones.
```
```suggestion
  description: Good corpsman glasses, can protect from flashes.
```

KillerTheFlareon replied:
Brigmedic glasses have the same description, I just extended the joke.

Rinary1 replied:
> Brigmedic glasses have the same description, I just extended the joke.

It's better to avoid such descriptions and jokes, they are local and may not be understood by new players.

KillerTheFlareon replied:
> > Brigmedic glasses have the same description, I just extended the joke.
> 
> It's better to avoid such descriptions and jokes, they are local and may not be understood by new players.

I don't really see the harm in it, that sort of thing was considered fine when the brigmedic glasses were first implemented and never changed. All it really does, if the person doesn't get the joke, is make them wonder who this "Tamashi" person is. Makes them ask questions and such, and then they get in on the joke because it was explained and they understand it now.

Rinary1 replied:
> > > Brigmedic glasses have the same description, I just extended the joke.
> > 
> > It's better to avoid such descriptions and jokes, they are local and may not be understood by new players.
> 
> I don't really see the harm in it, that sort of thing was considered fine when the brigmedic glasses were first implemented and never changed. All it really does, if the person doesn't get the joke, is make them wonder who this "Tamashi" person is. Makes them ask questions and such, and then they get in on the joke because it was explained and they understand it now.

Well, most likely the problem is that no one paid attention to this. I think that item descriptions should not contain jokes. Usually, when you read them, they should reflect the essence of the item, or the joke should be clear to everyone and associated with it.

KillerTheFlareon replied:
> > > > Brigmedic glasses have the same description, I just extended the joke.
> > > 
> > > 
> > > It's better to avoid such descriptions and jokes, they are local and may not be understood by new players.
> > 
> > 
> > I don't really see the harm in it, that sort of thing was considered fine when the brigmedic glasses were first implemented and never changed. All it really does, if the person doesn't get the joke, is make them wonder who this "Tamashi" person is. Makes them ask questions and such, and then they get in on the joke because it was explained and they understand it now.
> 
> Well, most likely the problem is that no one paid attention to this. I think that item descriptions should not contain jokes. Usually, when you read them, they should reflect the essence of the item, or the joke should be clear to everyone and associated with it.

In my opinion the joke _is_ pretty clear, as anyone who's been around SL long enough and got to see my face ends up hearing the tale about how I bugged one of the maintainers so much by playing brigmedic they ended up implementing the brigmedic as a role, heh. Then that tale gets told by other people, and so on. It's a nice little thing.

Rinary1 replied:
> In my opinion the joke _is_ pretty clear, as anyone who's been around SL long enough and got to see my face ends up hearing the tale about how I bugged one of the maintainers so much by playing brigmedic they ended up implementing the brigmedic as a role, heh. Then that tale gets told by other people, and so on. It's a nice little thing.

But we're talking about new players here, so there's no point dwelling on the old ones. If you want a pleasant memory, it's better to express it in a toy, statue, plushie, or similar items. But definitely not in the description of an default items.

KillerTheFlareon replied:
> > In my opinion the joke _is_ pretty clear, as anyone who's been around SL long enough and got to see my face ends up hearing the tale about how I bugged one of the maintainers so much by playing brigmedic they ended up implementing the brigmedic as a role, heh. Then that tale gets told by other people, and so on. It's a nice little thing.
> 
> But we're talking about new players here, so there's no point dwelling on the old ones. If you want a pleasant memory, it's better to express it in a toy, statue, plushie, or similar items. But definitely not in the description of an default items.

Fair enough I suppose.

KillerTheFlareon replied:
Changes made.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -27,7 +27,7 @@
   id: ClothingEyesHudBrigmedic
   name: brigmedic hud
-  description: Tamashi-upgraded HUDs. Yes. Brigmedic ones.
+  description: Tamashi-upgraded hud. Yes. Brigmedic ones.
```
```suggestion
  description: Brigmedic hud, differs from another ones in that it contains both medical and security information at the same time.
```

Isn't it's better to make general hud, like you do this with headset or like with mostly another huds?

KillerTheFlareon replied:
Color scheme differences - brigmedic hud clashes with the intended color scheme of corpsman

KillerTheFlareon replied:
Description changes made.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -47,6 +47,29 @@
+  parent: [ ClothingEyesBase, ShowSecurityIcons, BaseSecurityContraband ]
+  id: ClothingEyesHudCorpsman
+  name: corpsman hud
+  description: Super Tamashi-Upgraded hud. Corpsman one.
```
```suggestion
  description: Corpsman hud, differs from another ones in that it contains both medical and security information at the same time.
```

KillerTheFlareon replied:
As stated in other commits, extends the already existing joke.

KillerTheFlareon replied:
Changes made.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Markers/Spawners/jobs.yml:
```diff
@@ -105,3 +105,16 @@
+    sprite: _Starlight/Markers/jobs.rsi
+    layers:
+      - state: green
+      - state: corpsman
```
```suggestion
      - state: corpsman
      - state: jobicon
```

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/Medical/handheld_corpsman_crew_monitor.yml:
```diff
@@ -0,0 +1,37 @@
+  components:
+  - type: Tag
+    tags:
+      - BrigmedicBeltEquip
```
-> CorpsmanBeltEquip

KillerTheFlareon replied:
Corps uses the same belt equip as brig, there's no reason to make it a different tag

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/chemical-containers.yml:
Instead of making they indestructible, just make it has bigger destruction trigger.

KillerTheFlareon replied:
That goes against the entire point of them

KillerTheFlareon replied:
To further elaborate, they are intended to be indestructible. That's why it's locked behind research and uses advanced materials to make.

Rinary1 replied:
> That goes against the entire point of them

I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.

KillerTheFlareon replied:
> > That goes against the entire point of them
> 
> I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.

Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place

Rinary1 replied:
> > > That goes against the entire point of them
> > 
> > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> 
> Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place

This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be

KillerTheFlareon replied:
> > > > That goes against the entire point of them
> > > 
> > > 
> > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > 
> > 
> > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> 
> This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be

It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
The occasional maints find too, if a stack of it is mapped into maints.

If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place.

Rinary1 replied:
> > > > > That goes against the entire point of them
> > > > 
> > > > 
> > > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > > 
> > > 
> > > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> > 
> > This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be
> 
> It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
> The occasional maints find too, if a stack of it is mapped into maints.
> 
> If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place. 

Again, I don't think that the effort justifies the unbreakability and increased capacity. Materials can be a problem, I agree, but I believe that in any case, you can find a way around it. While playing the game, I often encountered more problems with research than with materials. Often, they don't reach t3-t4, so as a rule, the effort justifies the items (and sometimes the items don't justify the effort).

KillerTheFlareon replied:
> > > > > > That goes against the entire point of them
> > > > > 
> > > > > 
> > > > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > > > 
> > > > 
> > > > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > > > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> > > 
> > > 
> > > This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be
> > 
> > 
> > It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
> > The occasional maints find too, if a stack of it is mapped into maints.
> > If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place.
> 
> Again, I don't think that the effort justifies the unbreakability and increased capacity. Materials can be a problem, I agree, but I believe that in any case, you can find a way around it. While playing the game, I often encountered more problems with research than with materials. Often, they don't reach t3-t4, so as a rule, the effort justifies the items (and sometimes the items don't justify the effort).

We already have roundstart items that have increased capacity- granted it's a beaker, but you can pay for it with credits. I'm giving an in-round way to do that, without making it impossible to obtain. Which is why I didn't want to make it T3-T4 in the first place, actually. They would need a lot of reinforced glass to do an entire replacement of all the containers in medical, and in the meantime and before that antags can still destroy the non-research containers like before.

It's there to both provide some QoL to medical and make utterly destroying medical's ability to treat people a little harder than just throwing a minibomb at the medical locker, assuming medical takes the time out of their day to get sci to research it and then make the containers.

Rinary1 replied:
> > > > > > > That goes against the entire point of them
> > > > > > 
> > > > > > 
> > > > > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > > > > 
> > > > > 
> > > > > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > > > > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> > > > 
> > > > 
> > > > This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be
> > > 
> > > 
> > > It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
> > > The occasional maints find too, if a stack of it is mapped into maints.
> > > If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place.
> > 
> > Again, I don't think that the effort justifies the unbreakability and increased capacity. Materials can be a problem, I agree, but I believe that in any case, you can find a way around it. While playing the game, I often encountered more problems with research than with materials. Often, they don't reach t3-t4, so as a rule, the effort justifies the items (and sometimes the items don't justify the effort).
> 
> We already have roundstart items that have increased capacity- granted it's a beaker, but you can pay for it with credits. I'm giving an in-round way to do that, without making it impossible to obtain. Which is why I didn't want to make it T3-T4 in the first place, actually. They would need a lot of reinforced glass to do an entire replacement of all the containers in medical, and in the meantime and before that antags can still destroy the non-research containers like before.
> 
> It's there to both provide some QoL to medical and make utterly destroying medical's ability to treat people a little harder than just throwing a minibomb at the medical locker, assuming medical takes the time out of their day to get sci to research it and then make the containers.

I think you're right, but I'm not asking for them to be breakable with a single attack or explosion. I'm asking for them to be made so that they can be broken in some way using extremely radical methods, instead of, as you say, “throwing a mini-bomb into a locker.”

KillerTheFlareon replied:
> > > > > > > > That goes against the entire point of them
> > > > > > > 
> > > > > > > 
> > > > > > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > > > > > 
> > > > > > 
> > > > > > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > > > > > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> > > > > 
> > > > > 
> > > > > This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be
> > > > 
> > > > 
> > > > It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
> > > > The occasional maints find too, if a stack of it is mapped into maints.
> > > > If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place.
> > > 
> > > 
> > > Again, I don't think that the effort justifies the unbreakability and increased capacity. Materials can be a problem, I agree, but I believe that in any case, you can find a way around it. While playing the game, I often encountered more problems with research than with materials. Often, they don't reach t3-t4, so as a rule, the effort justifies the items (and sometimes the items don't justify the effort).
> > 
> > 
> > We already have roundstart items that have increased capacity- granted it's a beaker, but you can pay for it with credits. I'm giving an in-round way to do that, without making it impossible to obtain. Which is why I didn't want to make it T3-T4 in the first place, actually. They would need a lot of reinforced glass to do an entire replacement of all the containers in medical, and in the meantime and before that antags can still destroy the non-research containers like before.
> > It's there to both provide some QoL to medical and make utterly destroying medical's ability to treat people a little harder than just throwing a minibomb at the medical locker, assuming medical takes the time out of their day to get sci to research it and then make the containers.
> 
> I think you're right, but I'm not asking for them to be breakable with a single attack or explosion. I'm asking for them to be made so that they can be broken in some way using extremely radical methods, instead of, as you say, “throwing a mini-bomb into a locker.”

Well.. gimme an example to work with here. What do you want it to _be_ breakable by?

Rinary1 replied:
> > > > > > > > > That goes against the entire point of them
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I wouldn't say that this breaks their point of view in any way, but instead of being indestructible (after all, they can't be broken even with a nuke), make them difficult to destroy, so that in normal situations their strength is sufficient, but there are still moments when they break.
> > > > > > > 
> > > > > > > 
> > > > > > > Well that's the thing. There isn't supposed to _be_ a point where they break. The containers exist as minor upgrades (in capacity) over the non-research ones, and to survive the infamous medical locker bombings.
> > > > > > > Plenty of other items survive a nuclear blast- there's actually very little that even have a HP threshold in the first place
> > > > > > 
> > > > > > 
> > > > > > This is quite a big problem with the game, since in theory everything should be breakable (with the possible exception of admeme). In any case, even if they need to be researched, that doesn't mean it justifies the fact that they can't be
> > > > > 
> > > > > 
> > > > > It's not just the fact it needs research, but the fact it needs an advanced material as well. Reinforced glass, something that isn't easily obtainable asides from yelling at cargo (which costs money) or begging engineering.
> > > > > The occasional maints find too, if a stack of it is mapped into maints.
> > > > > If it were something roundstart, I'd agree, but it needs quite a bit of effort to obtain in the first place.
> > > > 
> > > > 
> > > > Again, I don't think that the effort justifies the unbreakability and increased capacity. Materials can be a problem, I agree, but I believe that in any case, you can find a way around it. While playing the game, I often encountered more problems with research than with materials. Often, they don't reach t3-t4, so as a rule, the effort justifies the items (and sometimes the items don't justify the effort).
> > > 
> > > 
> > > We already have roundstart items that have increased capacity- granted it's a beaker, but you can pay for it with credits. I'm giving an in-round way to do that, without making it impossible to obtain. Which is why I didn't want to make it T3-T4 in the first place, actually. They would need a lot of reinforced glass to do an entire replacement of all the containers in medical, and in the meantime and before that antags can still destroy the non-research containers like before.
> > > It's there to both provide some QoL to medical and make utterly destroying medical's ability to treat people a little harder than just throwing a minibomb at the medical locker, assuming medical takes the time out of their day to get sci to research it and then make the containers.
> > 
> > I think you're right, but I'm not asking for them to be breakable with a single attack or explosion. I'm asking for them to be made so that they can be broken in some way using extremely radical methods, instead of, as you say, “throwing a mini-bomb into a locker.”
> 
> Well.. gimme an example to work with here. What do you want it to _be_ breakable by?

Nuke or syndicate bomb(stationary), just so that there isn't a strange situation where they blow up the med and instead of the chemistry lab you see a pile of magic cans flying in the space that are stronger than the floor, walls, armored walls, and armored windows. And it will also solve the problem with parent.

KillerTheFlareon replied:
that is a really funny sight admittedly. Alright, i'll make the changes.

KillerTheFlareon replied:
Done. It will explode from syndie bombs and higher.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/chemistry-bottles.yml:
Instead of making they indestructible, just make it has bigger destruction trigger.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/chemistry-vials.yml:
Instead of making they indestructible, just make it has bigger destruction trigger.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/chemistry.yml:
Instead of making they indestructible, just make it has bigger destruction trigger.

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/stealTargetGroups.yml:
```diff
@@ -14,6 +14,24 @@
+
+# Syndie
+- type: stealTargetGroup
+  id: BrigmedicHypoSteal
```
```suggestion
  id: CorpsmanHypoSteal
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/stealTargetGroups.yml:
```diff
@@ -14,6 +14,24 @@
+# Syndie
+- type: stealTargetGroup
+  id: BrigmedicHypoSteal
+  name: steal-target-groups-brigmedichypo
```
```suggestion
  name: steal-target-groups-corpsman-hypo
```

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/Medical/hypospray.yml:
```diff
@@ -0,0 +1,90 @@
+  name: corpsman hypospray
+  parent: [BaseItem, BaseSecurityContraband]
+  description: A compact sterile spray for rapid chemical injection in combat situations. Accepts small vials.
+  id: BrigHypo
```
-> CorpsmanHypospray

KillerTheFlareon replied:
Done.

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Objects/Specific/Medical/hypospray.yml:
```diff
@@ -0,0 +1,90 @@
+    - HighRiskItem
+    - Hypospray
+  - type: StealTarget
+    stealGroup: BrigmedicHypoSteal
```
```suggestion
    stealGroup: CorpsmanHypoSteal
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+- type: entity
+  abstract: true
+  parent: BaseTraitorStealObjective
+  id: BaseBrigmedicStealObjective
```
```suggestion
  id: BaseCorpsmanStealObjective
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+  id: BaseBrigmedicStealObjective
+  components:
+  - type: NotJobRequirement
+    job: Brigmedic
```
```suggestion
    job: Corpsman
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+  - type: NotJobRequirement
+    job: Brigmedic
+  - type: StealCondition
+    owner: job-name-brigmedic
```
```suggestion
    owner: job-name-corpsman
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+    owner: job-name-brigmedic
+
+- type: entity
+  parent: BaseBrigmedicStealObjective
```
```suggestion
  parent: BaseCorpsmanStealObjective
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+
+- type: entity
+  parent: BaseBrigmedicStealObjective
+  id: BrigmedicHyposprayStealObjective
```
```suggestion
  id: CorpsmanHyposprayStealObjective
```

Rinary1 commented on Resources/Prototypes/_StarLight/Objectives/traitor.yml:
```diff
@@ -53,9 +53,31 @@
+  id: BrigmedicHyposprayStealObjective
+  components:
+  - type: StealCondition
+    stealGroup: BrigmedicHypoSteal
```
```suggestion
    stealGroup: CorpsmanHypoSteal
```

KillerTheFlareon committed:
Update Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Content.Shared/Chemistry/EntitySystems/SharedInjectorSystem.cs

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Locale/en-US/_Starlight/chemistry/hypospray-vial-loader.ftl

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Locale/en-US/_Starlight/chemistry/hypospray-vial-loader.ftl

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Locale/en-US/_Starlight/objectives/conditions/steal-target-groups.ftl

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
suggested changes

KillerTheFlareon committed:
Update Resources/Prototypes/Roles/Jobs/departments.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Entities/Markers/Spawners/jobs.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/stealTargetGroups.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/stealTargetGroups.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Entities/Objects/Specific/Medical/hypospray.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
Update Resources/Prototypes/_StarLight/Objectives/traitor.yml

Co-authored-by: Rinary <rinary.super@gmail.com>

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Medical-Additions-v2

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
changes

a

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
Requested changes

Still explosion proof for all intents and purposes but will not survive a syndie bomb

KillerTheFlareon committed:
anger

fixes

Rinary1 commented on Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs:
```diff
@@ -0,0 +1,90 @@
+                _prevVialEntity = null;
+            }
+
+            _label.SetMarkup(Loc.GetString("brigmedic-hypospray-volume-label",
```
```suggestion
            _label.SetMarkup(Loc.GetString("hypospray-nomode-volume-label",
```

KillerTheFlareon committed:
Update Content.Client/_Starlight/Chemistry/ExternalContainerInjector/ExternalContainerInjectorStatusControl.cs

Co-authored-by: Rinary <rinary.super@gmail.com>
